node/pegnet/transaction.go

```diff
@@ -0,0 +1,60 @@
+package pegnet
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+const (
+	DefaultPad = 3
+)
+
+// SplitTxID splits a TxID into it's parts.
+// TxID format : [TxIndex]-[EntryHash]
+//				 1-c99dedea0e4e0c40118fe7e4d515b23cc0489269c8cef187b4f15a4ccbd880be
+func SplitTxID(txid string) (index int, entryhash string, err error) {
+	arr := strings.Split(txid, "-")
+	if len(arr) != 2 {
+		return -1, "", fmt.Errorf("txid does not match txid format, format: [TxIndex]-[EntryHash]")
+	}
+
+	txIndex, err := strconv.ParseInt(arr[0], 10, 32)
+	if err != nil {
+		return -1, "", fmt.Errorf("index must be a valid integer")
+	}
+	if txIndex < 0 {
```

Comment:
this can't happen because if there's a "-", then strings.Split returns 3, and strconv.ParseInt can't overflow into negative (it throws an error)

Reply:
Yea, I noticed that in my unit tests. I can drop this check